crypto/fipsmodule/rand/new_rand_test.cc

```diff
@@ -103,6 +114,48 @@ TEST(NewRand, ReseedInterval) {
   ASSERT_TRUE(RAND_bytes(randomness, request_len_new_reseed));
   ASSERT_EQ(get_thread_reseed_calls_since_initialization(), reseed_calls_since_initialization + 2);
   ASSERT_EQ(get_thread_generate_calls_since_seed(), 1ULL);
+
+  *returnFlag = true;
+}
+
+TEST_F(newRandTest, ReseedIntervalWhenUbeIsSupported) {
+  if (!UbeIsSupported()) {
+    GTEST_SKIP() << "UBE detection is not supported";
+  }
+  ASSERT_TRUE(threadTest(10, randReseedIntervalUbeIsSupportedTests));
+}
+
+static void randReseedIntervalUbeNotSupportedTests(bool *returnFlag) {
+  uint8_t *randomness = (uint8_t *) OPENSSL_zalloc(CTR_DRBG_MAX_GENERATE_LENGTH);
+  bssl::UniquePtr<uint8_t> deleter(randomness);
+  uint64_t generate_calls_since_seed = get_thread_generate_calls_since_seed();
+  uint64_t reseed_calls_since_initialization = get_thread_reseed_calls_since_initialization();
+
+  if (kCtrDrbgReseedInterval - generate_calls_since_seed < 2) {
+    // Ensure the reseed interval doesn't conflict with logic below.
+    ASSERT_TRUE(RAND_bytes(randomness, 1));
+    ASSERT_TRUE(RAND_bytes(randomness, 1));
+  }
+
+  // Each invocation of the randomness generation induce a reseed due to UBE
+  // detection not being supported.
+  ASSERT_TRUE(RAND_bytes(randomness, 1));
+  ASSERT_EQ(get_thread_generate_calls_since_seed(), 1ULL);
+  ASSERT_EQ(get_thread_reseed_calls_since_initialization(), reseed_calls_since_initialization + 1);
+
+  ASSERT_TRUE(RAND_bytes(randomness, 1));
+  ASSERT_EQ(get_thread_generate_calls_since_seed(), 1ULL);
+  ASSERT_EQ(get_thread_reseed_calls_since_initialization(), reseed_calls_since_initialization + 2);
+
+  *returnFlag = true;
+}
+
+TEST_F(newRandTest, ReseedIntervalWhenUbeNotSupported) {
```

Comment:
Yes, since `ubeTest, ExpectedSupportTests` is testing for amazon linux, expects ube to be supported, and we have plenty of amazon linux in CI.
But otherwise, the tests can't ensure something is executed unless the operating environment supports it. That's an external thing.